File: java/src/main/java/com/carepet/model/Sensor.java
package com.carepet.model;

import com.datastax.oss.driver.api.mapper.annotations.ClusteringColumn;
import com.datastax.oss.driver.api.mapper.annotations.CqlName;
import com.datastax.oss.driver.api.mapper.annotations.Entity;
import com.datastax.oss.driver.api.mapper.annotations.PartitionKey;
import com.fasterxml.jackson.annotation.JsonProperty;
import org.apache.commons.lang.math.RandomUtils;

import java.util.UUID;

@Entity
@CqlName("sensor")
public class Sensor {
    @PartitionKey
    @JsonProperty("pet_id")
    private UUID petId;

    @ClusteringColumn
    @JsonProperty("sensor_id")
    private UUID sensorId;

    private String type;

    public Sensor() {
    }

    public Sensor(UUID petId, UUID sensorId, String type) {
        this.petId = petId;
        this.sensorId = sensorId;
        this.type = type;
    }

    public static Sensor random(UUID petId) {
        return new Sensor(
                petId,
                UUID.randomUUID(),
                SensorType.values()[RandomUtils.nextInt(SensorType.values().length)].getType());
    }

    public UUID getPetId() {
        return petId;
    }

    public void setPetId(UUID petId) {
        this.petId = petId;
    }

    public UUID getSensorId() {
        return sensorId;
    }

    public void setSensorId(UUID sensorId) {
        this.sensorId = sensorId;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public float randomData() {
        switch (SensorType.fromString(type)) {
            case Temperature:
                // average F
                return 101.0f + RandomUtils.nextInt(10) - 4;
            case Pulse:
                // average beat per minute
                return 100.0f + RandomUtils.nextInt(40) - 20;
            case Respiration:
                // average inhales per minute
                return 35.0f + RandomUtils.nextInt(5) - 2;
            case Location:
                // pet can teleport
                return 10 * RandomUtils.nextFloat();
            default:
                return 0.0f;
        }
    }

    @Override
    public String toString() {
        return "Sensor{" +
                "petId=" + petId +
                ", sensorId=" + sensorId +
                ", type='" + type + '\'' +
                '}';
    }
}

File: csharp/Model/Sensor.cs
using System;
using Cassandra.Mapping.Attributes;
using Newtonsoft.Json;

namespace CarePet.Model
{
    [Table(Name = "sensor")]
    public class Sensor
    {
        [PartitionKey]
        [JsonProperty("pet_id")]
        [Column("pet_id")]
        public Guid PetId { get; set; }

        [ClusteringKey]
        [JsonProperty("sensor_id")]
        [Column("sensor_id")]
        public Guid SensorId { get; set; }

        public string Type { get; set; }

        public Sensor()
        {
        }

        public Sensor(Guid petId, Guid sensorId, string type)
        {
            PetId = petId;
            SensorId = sensorId;
            Type = type;
        }

        public static Sensor Random(Guid petId)
        {
            var types = Enum.GetValues(typeof(SensorType));
            var random = new Random();
            var randomType = (SensorType)types.GetValue(random.Next(types.Length));
            var typeCode = SensorTypeExtensions.GetTypeCode(randomType);
            return new Sensor(petId, Guid.NewGuid(), typeCode);
        }

        public float RandomData()
        {
            var random = new Random();
            switch (SensorTypeExtensions.FromString(Type))
            {
                case SensorType.Temperature:
                    // average F
                    return 101.0f + random.Next(10) - 4;
                case SensorType.Pulse:
                    // average beats per minute
                    return 100.0f + random.Next(40) - 20;
                case SensorType.Respiration:
                    // average inhales per minute
                    return 35.0f + random.Next(5) - 2;
                case SensorType.Location:
                    // pet can teleport
                    return (float)(10 * random.NextDouble());
                default:
                    return 0.0f;
            }
        }

        public override string ToString()
        {
            return $"Sensor{{petId={PetId}, sensorId={SensorId}, type='{Type}'}}";
        }
    }
}
